src/Microsoft.OpenApi/Services/OpenApiUrlTreeNode.cs

```diff
@@ -150,6 +150,13 @@ public void Attach(OpenApiDocument doc, string label)
             }
 
             var segments = path.Split('/');
+            if (path.EndsWith("/"))
+            {
+                // Remove the last element, which is empty, and append the trailing slash to the new last element
```

Comment:
I'm a bit worried that this will result in additional "empty" nodes that were not here previously and would potentially. Can you add unit tests to control for that please?

Reply:
I've just pushed a commit to augment the tests and make sure that the leaf nodes, which have a path with a trailing slash, have no children.

Let me know if there's other cases we should cover.